src/System.IO.FileSystem/src/System/IO/WinRTFileSystem.cs

```diff
@@ -227,14 +228,86 @@ public override bool DirectoryExists(string fullPath)
 
         public override IEnumerable<string> EnumeratePaths(string fullPath, string searchPattern, SearchOption searchOption, SearchTarget searchTarget)
         {
-            // Fill in implementation
-            return new string[0];
+            IEnumerable<IStorageItem> storageFiles = SynchronousResultOf(EnumerateFileQuery(fullPath, searchPattern, searchOption, searchTarget));
```

Comment:
I think there's a subtle difference between this implementation and the Win32 and Unix implementations.  Because this is implemented as an iterator, calling this method (same feedback goes for the other methods below) isn't going to actually start enumerating... that won't happen until MoveNext is called.  As such, any errors that result from starting to enumerate won't propagate until the first call to MoveNext rather than propagating out of the synchronous call to Enumerate*.